public/app/plugins/datasource/loki/datasource.test.ts

```diff
@@ -1198,11 +1198,43 @@ describe('LokiDatasource', () => {
   });
 
   describe('getQueryStats', () => {
+    let ds: LokiDatasource;
+    beforeEach(() => {
+      ds = createLokiDatasource(templateSrvStub);
+      ds.statsMetadataRequest = jest.fn().mockResolvedValue({ streams: 1, chunks: 1, bytes: 1, entries: 1 });
+      ds.interpolateString = jest.fn().mockImplementation((value: string) => value.replace('$__interval', '1m'));
+    });
+
     it('uses statsMetadataRequest', async () => {
-      const ds = createLokiDatasource(templateSrvStub);
-      const spy = jest.spyOn(ds, 'statsMetadataRequest').mockResolvedValue({} as QueryStats);
-      ds.getQueryStats('{foo="bar"}');
-      expect(spy).toHaveBeenCalled();
+      const result = await ds.getQueryStats('{foo="bar"}');
+
+      expect(ds.statsMetadataRequest).toHaveBeenCalled();
+      expect(result).toEqual({ streams: 1, chunks: 1, bytes: 1, entries: 1 });
+    });
+
+    it('supports queries with template variables', async () => {
+      const result = await ds.getQueryStats('rate({instance="server\\1"}[$__interval])');
+
+      expect(result).toEqual({
+        streams: 1,
+        chunks: 1,
+        bytes: 1,
+        entries: 1,
+      });
+    });
+
+    it('does not call stats if the query is invalid', async () => {
+      const result = await ds.getQueryStats('rate({label="value"}');
+
+      expect(ds.statsMetadataRequest).not.toHaveBeenCalled();
+      expect(result).toBe(undefined);
+    });
+
+    it('combines the stats of each label matcher', async () => {
```

Comment:
@gwdawson Just saw that there was missing coverage for this part of the code:

https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/datasource.ts#L459-L463

Is that correct? Should you accumulate the stats of `foo => bar` and `test => test` in a query like `count_over_time({foo="bar"}[1m]) + count_over_time({test="test"}[1m])`?